### PR TITLE
No need to link explicitly with libpthread on Solaris

### LINKED ISSUE #19607: No need to link explicitly with libpthread on Solaris
Starting with Solaris 10 (released in Jan 2005) libpthread is a filter library over libc and all Solaris processes are always multi-threaded (even if they often only contain a single thread).  As such it has not been necessary to pass any information to the compiler or linker about libpthread via -lpthread or -pthreads.  This is independent of compiler (gcc, Studio or LLVM).

This means at any Illumos/OpenSolaris derived releases also don't need to explicitly link with libpthread.

Solaris 10 is the oldest release of Solaris still getting fixes from Oracle, Solaris 9 for example hasn't had any fixes since 2014 so there should be no need to support such an old OS release in OpenSSL 3.x

Oracle will be applying this patch to the OpenSSL 3.x builds included in Solaris 11.4

[050-lpthread.patch.txt](https://github.com/openssl/openssl/files/9938068/050-lpthread.patch.txt)

### PR DESCRIPTION
**Pull request by darrenmoffat**
CLA: trivial

Reviewed-by: Zdenek.Kotal@oracle.com
Reviewed-by: Ali.Bahrami@oracle.com

Fixes #19607

### CODE REVIEW

**Review by t8m — APPROVED** [review]
Given this is mostly removals, it seems acceptable to me with CLA:trivial.

File: Configurations/10-main.conf [hunk 1]
```diff
@@ -239,9 +239,8 @@ my %targets = (
         CFLAGS           => add_before(picker(default => "-Wall",
                                               debug   => "-O0 -g",
                                               release => "-O3 -fomit-frame-pointer")),
-        cflags           => add(threads("-pthread")),
+        cflags           => add("-m64"),
```
Comment (tmshort) [inline Configurations/10-main.conf:242]: Adding `-m64` to an x86 target?

**Review by tmshort — CHANGES_REQUESTED** [review]

File: Configurations/10-main.conf (hunk 1 above)
Comment (darrenmoffat) [reply Configurations/10-main.conf]: That was a mistake I've removed that now.

**Review by tmshort — APPROVED** [review]
LGTM

**Comment by tmshort** [comment]
This is primarily deletes (with one edit). I don't know if it qualifies for CLA:trivial, but I'd be OK with it.

**Review by t8m — APPROVED** [review]
OK with CLA: trivial

**Review by Bunos07 — APPROVED** [review]

**Pull request closed without merging**